### PR TITLE
Parse and plot powmon power data 

### LINKED ISSUE #415: Parse and plot powmon power data
- Create a script to plot CSV power data (node, GPUs, CPUs, memory) obtained from powmon, given a powmon data file. 
- Provide an option in the script to look at multi node aggregated data. 

@altahat2003 is working on this.

### PR DESCRIPTION
**Pull request by altahat2003**
# Description

This script plots power data of `powmon`.

Fixes #415 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature/architecture support (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Build/CI update

# How Has This Been Tested?

This 

- [ ] Test A: Hardware architecture, machine name, example/test run
- [ ] Test B: Hardware architecture, machine name, example/test run
- [x] Tested on two powmon runs against AMG app

# Checklist:

- [x] I have run `./scripts/check-code-format.sh` and confirm my code code follows the style guidelines of variorum
- [ ] I have added comments in my code
- [ ] My changes generate no new warnings (build with `-DENABLE_WARNINGS=ON`)
- [ ] New and existing unit tests pass with my changes

Thank you for taking the time to contribute to Variorum!

### CODE REVIEW

**Comment by tpatki** [comment]
Hi @altahat2003 : thank you for this! Somehow it looks like you didn't see the PR template checklist, not sure why. Can you run the `scripts/check-code-format.sh` and push up the reformatted file, so the code is formatted correctly?

**Comment by tpatki** [comment]
Hi @altahat2003 LGTM! I'm going to test once at my end to get a feel for it, and then merge. I think it can go in v0.7 -- thank you for your help with this useful script!

**Comment by altahat2003** [comment]
Hi @tpatki, thanks for your help and guidance to get my first PR submitted successfully :)

File: src/powmon/scripts/powmon-plot.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+# Options:
+#   --input or -i: required to specify power data path. Make sure the power data in the
+#     the directory is from a single run.
+#   --output or -o: optional to specify where to save the plots. If not defined, the input path
+#     is used.
+#   --type or -t: optional to select what to plot.
+#     per-node: plot power data per node only. If the app were run on 4 nodes, 4 plots
+#     are created.
+#     aggregate: plot discriptive stats of the power data of all nodes. There should be four plots, mean, max, min, median.
+#
```
Comment (tpatki) [inline src/powmon/scripts/powmon-plot.py:28]: Add the `-d description` flag here.

File: src/powmon/scripts/powmon-plot.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        type=str,
+        required=False,
+        help="Output directory path. If not defined, plots will be saved in the input path.",
+    )
+    parser.add_argument(
+        "--description",
+        "-d",
+        type=str,
+        required=True,
+        help="Output directory path. If not defined, plots will be saved in the input path.",
```
Comment (tpatki) [inline src/powmon/scripts/powmon-plot.py]: I believe this sets the title, please correct the help message here.

**Review by tpatki — CHANGES_REQUESTED** [review]

File: src/powmon/scripts/powmon-plot.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+# file holds power data of a node run the tasks. It plots power data per node, where there is a figure for each node.
+#  It also plots the descriptive stats of the power data of all nodes.
+# Requirements:
+#   pandas>=1.5.0
+#   numpy>=1.23.3
+#   matplotlib>=3.6.0
+#   argparse>=1.4.0
+#
+# How to run the script?
+#   python3 plotting.py [options]
```
Comment (tpatki) [inline src/powmon/scripts/powmon-plot.py]: Name should be `powmon-plot.py`

File: src/powmon/scripts/powmon-plot.py [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+        min.append(df[header].min())
+        median.append(df[header].median())
+    return (mean, min, max, median)
+
+
+# ---------------------------------------------------------------------
+# main
+# ---------------------------------------------------------------------
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(prog="plotting", description="Plotting Power Data")
```
Comment (tpatki) [inline src/powmon/scripts/powmon-plot.py]: `prog="powmon_plot"` here.

Comment (altahat2003) [reply src/powmon/scripts/powmon-plot.py]: It has been updated!

**Review by tpatki — DISMISSED** [review]

**Comment by tpatki** [comment]
@slabasan @altahat2003 

Two finals thing left for this PR (updating here in case I'm not around tomorrow, also see slack). 
- [ ] Add a check for python3.8 and print out an error if it is an earlier version.
- [ ] Fix x-axis on aggregated plot


I've tested it thoroughly on Lassen otherwise, and it looks good. Mostly ready to go, I think it's good to add it to v0.7 if we can get it all done in time today. If not, feel free to push it out to v0.8. 

One known issue is when GPU/Mem power is reported as -1 (eg on Quartz or Tioga), the plot data may look messed up in that case. This isn't a problem with the script, but rather with data being unavailable on some platforms. We can handle this issue later by just not plotting data that has -1 (or aggregated negative) values, I can open a bug fix, should be quick to update.

**Review by slabasan — APPROVED** [review]

**Pull request merged**